discovery.go

```diff
@@ -29,7 +64,10 @@ func init() {
 
 func main() {
 	log.SetFlags(0)
-	handling.Setup()
+	viper.Set("etcd", mustHostOnlyURL(viper.GetString("etcd")))
```

Comment:
it doesn't look like anything in the codebase will do `GetString("etcd")` after this, so no need to set it?